### DISCUSSION TITLE
ESP32 and GC9A01 IPS LCD 240x240 = watchdog reset

### DISCUSSION THREAD

**Discussion by Deadeye5589** [moononournation/Arduino_GFX#403] [Q&A]
Hello,

I have been trying to use the Arduino_GFX library on an ESP32 to run one of those round 240x240 LCD screens with the GC9A01 chipset. But I all can archive is a permanent boot loop, with the watchdog as the reset source. 

The ESP32 board is an AZ Delivery Wroom32. According to their manual, this should be an Wroom 32 chipset. They also recommend to select ESP32 Dev Board in the Arduino IDE as Target. After some testing I have stripped down the Hello World example to the point where on the screen color is set to black. Still the bootloop occours. Hmm, maybe wrong board definition then? So I tested some other ESP32 Targets as well. Same result, all lead to an bootloop. 

Does anybody know what I have might been doing wrong? 

The only changes I made to the Hello World example are changeing Date Bus and Chip to my chipset. Selecting ESP32 Dev  Boad and target and let the Arduino IDE compile everything. All bibs, board definitions and the IDE are up to date. 

Arduino_DataBus *bus = new Arduino_ESP32SPI(27 /* DC */, 5 /* CS */, 18 /* SCK */, 23 /* MOSI */, GFX_NOT_DEFINED /* MISO */, VSPI /* spi_num */);
Arduino_GFX *gfx = new Arduino_GC9A01(  bus, 7 /* RST */, 0 /* rotation */, true /* IPS */);

**Comment by rdekens** [comment]
for the 2.8" lilygo T--RGB i use 
![image](https://github.com/moononournation/Arduino_GFX/assets/12215022/e82446a6-7fd1-49ff-99c7-542124e60c33)

at least as a tarting point

**Comment by Deadeye5589** [comment]
I see that you are using a S3 ESP32. Mine is just a regular ESP32, hence the data bus would be different too

**Comment by Deadeye5589** [comment]
Those are the settings given by the Arduino IDE for an ESP32 Dev Board

![grafik](https://github.com/moononournation/Arduino_GFX/assets/22077414/85b88c29-df16-4b39-bd30-e500f058c5ef)

This is the whole source code I'm using

` #include <Arduino_GFX_Library.h>

#define GFX_BL DF_GFX_BL // default backlight pin, you may replace DF_GFX_BL to actual backlight pin

/* More dev device declaration: https://github.com/moononournation/Arduino_GFX/wiki/Dev-Device-Declaration */
#if defined(DISPLAY_DEV_KIT)
Arduino_GFX *gfx = create_default_Arduino_GFX();
#else /* !defined(DISPLAY_DEV_KIT) */

/* More data bus class: https://github.com/moononournation/Arduino_GFX/wiki/Data-Bus-Class */
Arduino_DataBus *bus = new Arduino_ESP32SPI(27 /* DC */, 5 /* CS */, 18 /* SCK */, 23 /* MOSI */, GFX_NOT_DEFINED /* MISO */, VSPI /* spi_num */);

/* More display class: https://github.com/moononournation/Arduino_GFX/wiki/Display-Class */
Arduino_GFX *gfx = new Arduino_GC9A01(  bus, 7 /* RST */, 0 /* rotation */, true /* IPS */);

#endif /* !defined(DISPLAY_DEV_KIT) */
/*******************************************************************************
 * End of Arduino_GFX setting
 ******************************************************************************/

void setup(void)
{
  Serial.begin(115200);
  // Serial.setDebugOutput(true);
  // while(!Serial);
  Serial.println("Arduino_GFX Hello World example");

#ifdef GFX_EXTRA_PRE_INIT
  GFX_EXTRA_PRE_INIT();
#endif

  // Init Display
  if (!gfx->begin())
  {
    Serial.println("gfx->begin() failed!");
  }
  gfx->fillScreen(BLACK);

  Serial.println("Hello World!");

  delay(5000); // 5 seconds
}

void loop()
{
  Serial.println("Hello World!");

  delay(1000); // 1 second
}`

In the serial monitor I can see that Arduino_GFX Hello World example is printed out and afterwards the watchdog hits. 

I also gave it a try and changed the display class to

`Arduino_GFX *gfx = new Arduino_ST7789(  bus, 7 /* RST */, 0 /* rotation */, true /* IPS */);`

still bootloop. Henche I would assume that the problem lies within the SPI definition in combination with the ESP32.

So next try, this time using the HSPI instead of VSPI.

`Arduino_DataBus *bus = new Arduino_ESP32SPI(27 /* DC */, 5 /* CS */, 14 /* SCK */, 13 /* MOSI */, GFX_NOT_DEFINED /* MISO */, HSPI /* spi_num */); `

same reusult. Something is broken with the ESP32SPI class and I don't know what it could be

**Reply by moononournation** [reply]
are you using dev version arduino-esp32? i also encounter this before.

**Reply by Deadeye5589** [reply]
More or less, yes. The chipset is an ESP32 Wroom 32. The board is a "white label" of the ESP32 dev kits. The schematic is identical to the Espressiv Development Boards with the Wroom 32 chipset. Hence the supplier recommends to set the Arduino IDE to "ESP32 Dev Board" as target.
